akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -597,36 +601,42 @@ class ClusterSingletonManager(singletonProps: Props, terminationMessage: Any, se
       getNextOldestChanged()
       stay
 
-    case Event(InitialOldestState(oldestOption, safeToBeOldest), _) =>
+    case Event(InitialOldestState(oldest, safeToBeOldest), _) =>
       oldestChangedReceived = true
-      if (oldestOption == selfUniqueAddressOption && safeToBeOldest)
+
+      if (oldest.headOption == selfUniqueAddressOption && safeToBeOldest)
         // oldest immediately
         tryGotoOldest()
-      else if (oldestOption == selfUniqueAddressOption)
-        goto(BecomingOldest).using(BecomingOldestData(None))
+      else if (oldest.headOption == selfUniqueAddressOption)
+        goto(BecomingOldest).using(BecomingOldestData(oldest.filterNot(_ == cluster.selfUniqueAddress)))
       else
-        goto(Younger).using(YoungerData(oldestOption))
+        goto(Younger).using(YoungerData(oldest.filterNot(_ == cluster.selfUniqueAddress)))
   }
 
   when(Younger) {
-    case Event(OldestChanged(oldestOption), YoungerData(previousOldestOption)) =>
+    case Event(OldestChanged(oldestOption), YoungerData(previousOldest)) =>
       oldestChangedReceived = true
       if (oldestOption == selfUniqueAddressOption) {
-        logInfo("Younger observed OldestChanged: [{} -> myself]", previousOldestOption.map(_.address))
-        previousOldestOption match {
-          case None                                 => tryGotoOldest()
-          case Some(prev) if removed.contains(prev) => tryGotoOldest()
-          case Some(prev) =>
-            peer(prev.address) ! HandOverToMe
-            goto(BecomingOldest).using(BecomingOldestData(previousOldestOption))
+        logInfo("Younger observed OldestChanged: [{} -> myself]", previousOldest.headOption.map(_.address))
+        if (previousOldest.isEmpty)
+          tryGotoOldest()
+        else if (previousOldest.forall(removed.contains))
+          tryGotoOldest()
```

Comment:
Perhaps fold this into a single `if (previousOldest.isEmpty || previousOldest.forall(removed.contains)) tryGotoOldest else handover` - could also be expressed as a single `previousOldest.forall(removed.contains)`

Reply:
fixed, changed to single forall